opal/mca/common/ofi/common_ofi.c

```diff
@@ -482,10 +482,16 @@ static int compute_dev_distances(pmix_device_distance_t **distances,
     if (PMIX_SUCCESS != ret) {
         goto out;
     }
+    /* if we are not bound, then we cannot compute distances */
+    if (hwloc_bitmap_iszero(cpuset.bitmap) ||
+        hwloc_bitmap_isfull(cpuset.bitmap)) {
+        return OPAL_ERR_NOT_BOUND;
+    }
 
```

Comment:
For some reason I got a compile error within internal hwloc and had to apply this patch
```
diff --git a/opal/mca/common/ofi/common_ofi.c b/opal/mca/common/ofi/common_ofi.c
index 6e3db480e0..a31ec779f5 100644
--- a/opal/mca/common/ofi/common_ofi.c
+++ b/opal/mca/common/ofi/common_ofi.c
@@ -499,7 +499,7 @@ static int compute_dev_distances(pmix_device_distance_t **distances,
     ninfo = 1;
     info = PMIx_Info_create(ninfo);
     PMIx_Info_load(&info[0], PMIX_DEVICE_TYPE, &type, PMIX_DEVTYPE);
-    ret = PMIx_Compute_distances(pmix_topo, &cpuset, info, ninfo, distances,
+    ret = PMIx_Compute_distances(&pmix_topo, &cpuset, info, ninfo, distances,
                                  ndist);
     PMIx_Info_free(info, ninfo);
```
or compile fails
```
common_ofi.c: In function 'compute_dev_distances':
common_ofi.c:502:34: error: incompatible type for argument 1 of 'PMIx_Compute_distances'
     ret = PMIx_Compute_distances(pmix_topo, &cpuset, info, ninfo, distances,
                                  ^~~~~~~~~
In file included from ../../../../opal/mca/pmix/pmix-internal.h:47:0,
                 from ../../../../opal/util/proc.h:28,
                 from common_ofi.h:21,
                 from common_ofi.c:31:
/home/ec2-user/ompi/3rd-party/openpmix/include/pmix.h:1062:27: note: expected 'pmix_topology_t * {aka struct <anonymous> *}' but argument is of type 'pmix_topology_t {aka struct <anonymous>}'
 PMIX_EXPORT pmix_status_t PMIx_Compute_distances(pmix_topology_t *topo,
                           ^~~~~~~~~~~~~~~~~~~~~~
```

But after applying the patch and build ompi, I got the same segfault running osu_mbw_mr with `--bind-to none` - `--bind-to core` works though.
```
#0  0x00007fae5020b707 in __strncasecmp_l_avx () from /lib64/libc.so.6
#1  0x00007fae4f2f899c in pmix_hwloc_load_topology () from /home/ec2-user/ompi/install/lib/libpmix.so.2
#2  0x00007fae4f31a23d in _loadtp () from /home/ec2-user/ompi/install/lib/libpmix.so.2
#3  0x00007fae4ea67b59 in event_process_active_single_queue (base=base@entry=0xfdd070, activeq=0xfdd4c0, max_to_process=max_to_process@entry=2147483647, 
    endtime=endtime@entry=0x0) at event.c:1691
#4  0x00007fae4ea682e7 in event_process_active (base=0xfdd070) at event.c:1783
#5  event_base_loop (base=0xfdd070, flags=1) at event.c:2006
#6  0x00007fae4f35a3f4 in progress_engine () from /home/ec2-user/ompi/install/lib/libpmix.so.2
#7  0x00007fae5047444b in start_thread () from /lib64/libpthread.so.0
#8  0x00007fae501af52f in clone () from /lib64/libc.so.6
```

Clues?